src/Aspire.Cli/Packaging/PackagingService.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Cli.NuGet;
+
+namespace Aspire.Cli.Packaging;
+
+internal interface IPackagingService
+{
+    public Task<IEnumerable<PackageChannel>> GetChannelsAsync(CancellationToken cancellationToken = default);
+}
+
+internal class PackagingService(CliExecutionContext executionContext, INuGetPackageCache nuGetPackageCache) : IPackagingService
+{
+    public Task<IEnumerable<PackageChannel>> GetChannelsAsync(CancellationToken cancellationToken = default)
+    {
+        var defaultChannel = PackageChannel.CreateImplicitChannel(nuGetPackageCache);
+        var stableChannel = PackageChannel.CreateExplicitChannel("stable", new[] 
+        {
+            new PackageMapping(PackageMapping.AllPackages, "https://api.nuget.org/v3/index.json")
+        }, nuGetPackageCache);
+
+        var dailyChannel = PackageChannel.CreateExplicitChannel("daily", new[]
```

Comment:
Each channel can have multiple mappings. Because we also need to specify package source mappings in the nuget config feeds might be repesented multiple times. When we write temporary nuget.config files we dedupe them in the package sources list.